lib/external_service_new/zendesk_downloader.rb

```diff
@@ -0,0 +1,51 @@
+module ExternalServiceNew
+	class ZendeskDownloader
+		def initialize(dispatcher:, logger:nil)
+			@dispatcher = dispatcher
+			@logger			= logger || respond_to?(:logger) ? logger : nil
+		end
+
+		def each_list(&block)
+			return to_enum(:each_list) unless block
+			[[
+				dispatcher.site_url,
+				"#{dispatcher.site_url}'s customers"
+			]].each(&block)
+		end
+
+		def each_email(id: nil, &block)
+			ZendeskInternalDownloader.new(
+				dispatcher: dispatcher, 
+				logger: logger
+			).each_email(&block)
+		end
+
+		private
+
+		attr_reader :dispatcher, :logger
+	end
+
+	class ZendeskInternalDownloader
+		def initialize(dispatcher:, filters: nil, logger:nil)	
+			raise "dispatcher must not be nil, found #{dispatcher.inspect}" unless dispatcher
+
+			@dispatcher = dispatcher
+			@filters = filters
+			@logger = logger || respond_to?(:logger) ? logger : nil
+		end
+
+		def each_email(&block)
+			return to_enum(:each_email) unless block
+			logger && logger.info("Downloading Zendesk Customer List with #{filters ? filters.inspect : "no filters"}")	
+		
+			response = dispatcher.dispatch(:query, "type:user role:end-user")
+			response.each {|email| yield email["email"]}
+
+			logger && logger.info("Downloaded Zendesk Customer List")
```

Comment:
It's a fake one. We discuss this with @AlexCppns. 
Because a user can be created outside a list we don't get the lists list